contracts/debt/IncurDebtV1.sol

```diff
@@ -0,0 +1,305 @@
+            "_ohmAmount more than borrower limit"
+        );
+        require(_ohmAmount <= getAvailableToBorrow(), "_ohmAmount more than available loan");
+
+        borrower[msg.sender].debt += _ohmAmount;
+        totalOutstandingDebt += _ohmAmount;
+
+        ITreasury(treasury).incurDebt(_ohmAmount, OHM);
+        IERC20(OHM).safeTransfer(msg.sender, _ohmAmount);
+    }
+
+    /**
+     * @notice withdraws gOHM/sOHM  to _to address
+     * - msg.sender must be a borrower
+     * - _amount (in OHM) must be less than or equal to depositedOhm - debt
+     * @dev should approve the staking contract to spend users sOHM
+     * @param _amount amount of gOHM/sOHM to withdraw
+     * @param _to address to send gOHM/sOHM
+     * @param _token token to send _amount
+     */
+    function withdraw(
+        uint256 _amount,
+        address _to,
+        address _token
+    ) external isBorrower(msg.sender) {
+        require(_amount > 0, "invalid number");
+
+        updateCollateralInSOHM(msg.sender);
+
+        if (_token == gOHM) {
+            uint256 _sAmount = IgOHM(gOHM).balanceFrom(_amount);
+
+            require(
+                _sAmount <= borrower[msg.sender].collateralInSOHM - borrower[msg.sender].debt,
+                "_amount above available gOHM"
+            );
+            borrower[msg.sender].collateralInSOHM -= _sAmount;
+            borrower[msg.sender].collateralInGOHM -= _amount;
+
+            IStaking(staking).wrap(_to, _sAmount);
+        } else if (_token == sOHM) {
+            require(
+                _amount <= borrower[msg.sender].collateralInSOHM - borrower[msg.sender].debt,
+                "_amount above available sOHM"
+            );
+
+            borrower[msg.sender].collateralInSOHM -= _amount;
+            borrower[msg.sender].collateralInGOHM -= IgOHM(gOHM).balanceTo(_amount);
+
+            IERC20(sOHM).safeTransfer(_to, _amount);
+        }
+    }
+
+    /**
+     * @notice repay debt with collateral
+     * - msg.sender must be a borrower
+     * - msg.sender's OHM allowance must be >= _ohmAmount
+     */
+    function repayDebtWithCollateral() external {
+        uint256 depositedCollateralAfterRepay = _repay(msg.sender);
+        totalOutstandingDebt -= borrower[msg.sender].debt;
+
+        borrower[msg.sender].collateralInSOHM = depositedCollateralAfterRepay;
+        borrower[msg.sender].collateralInGOHM = IgOHM(gOHM).balanceTo(depositedCollateralAfterRepay);
+
+        borrower[msg.sender].debt = 0;
+    }
+
+    /**
+     * @notice repay debt with collateral and withdraw the leftover
+     * - msg.sender must be a borrower
+     * - msg.sender's OHM allowance must be >= _ohmAmount
+     */
+    function repayDebtWithCollateralAndWithdrawTheRest(address _tokenToReceiveExcess) external {
+        uint256 depositedCollateralAfterRepay = _repay(msg.sender);
+
+        if (depositedCollateralAfterRepay > 0) {
+            if (_tokenToReceiveExcess == sOHM) {
```

Comment:
_repay() unstakes sOHM into OHM. Following lines transfer sOHM or wrap into gOHM, but you have OHM now?

Reply:
No, I don't because after unstaking into OHM, the OHM debt is paid to the treasury here 
`ITreasury(treasury).repayDebtWithOHM(borrower[_addr].debt);`

The sOHM or wrap into gOHM is the users funds when interacting with the deposit function.

Reply:
🤦‍♂️ yep, my b